arch/riscv/Kconfig

```diff
@@ -112,6 +112,12 @@ config RISCV_SOC_HAS_CUSTOM_SYS_IO
 	  the RISC-V SoC needs to do something different and more than reading and
 	  writing the registers.
 
+config RISCV_SOC_HAS_CUSTOM_PER_CORE_INIT
```

Comment:
this should not be riscv specific, the same pattern is use elsewhere, maybe such customization is used here first, but IMO this should be generalized to accomodate future usage by other arches, so I suggest doing this the same way we did it in `include/zephyr/platform/hooks.h` and use a similar pattern.